core/src/main/java/org/apache/gravitino/storage/relational/JDBCBackend.java

```diff
@@ -569,6 +593,103 @@ public <E extends Entity & HasIdentifier> List<E> updateEntityRelations(
+              List<String> statsToDelete =
+                  deleteRelations.stream()
+                      .map(relation -> relation.getDestIdent().name())
+                      .collect(Collectors.toList());
+
+              deleteRelations.forEach(
+                  relation ->
+                      Preconditions.checkArgument(
+                          relation.getDestType() == STATISTIC,
+                          "The destination type of relation must be STATISTIC, but got %s",
+                          relation.getDestType()));
+
+              deleteCount.addAndGet(
+                  StatisticMetaService.getInstance()
+                      .batchDeleteStatisticPOs(
+                          sourceVertex.getLeft(), sourceVertex.getRight(), statsToDelete));
+            });
+        return deleteCount.get();
+      default:
+        throw new IllegalArgumentException(
+            String.format("Doesn't support the relation type %s", relType));
+    }
+  }
+
+  @Override
+  public <E extends Entity & HasIdentifier> Void insertEntitiesAndRelations(
+      Type relType, List<Entity.RelationalEntity<E>> entities, boolean overwrite)
+      throws IOException {
+    switch (relType) {
+      case METADATA_OBJECT_STAT_REL:
+        Preconditions.checkArgument(
+            overwrite, "The overwrite must be true for metadata object stats relation");
+
+        StatisticMetaService metaService = StatisticMetaService.getInstance();
+        List<StatisticEntity> statisticEntities = Lists.newArrayList();
+
+        Set<NameIdentifier> relatedIdents = Sets.newHashSet();
+        Set<Entity.EntityType> relatedEntityTypes = Sets.newHashSet();
+
+        for (Entity.RelationalEntity<E> relEntity : entities) {
+          Preconditions.checkArgument(
+              relEntity.relatedNameIdentifiers().size() == 1,
+              "Each entity must have exactly one related identifier");
+          NameIdentifier relatedIdent = relEntity.relatedNameIdentifiers().get(0);
+          relatedIdents.add(relatedIdent);
+          relatedEntityTypes.add(relEntity.relatedEntityType());
+          Preconditions.checkArgument(
+              relEntity.entity() instanceof StatisticEntity,
+              "The entity must be a StatisticEntity");
+          statisticEntities.add((StatisticEntity) relEntity.entity());
+        }
+
+        Preconditions.checkArgument(
+            relatedIdents.size() == 1,
+            "All entities must have the same related identifier, but got %s",
+            relatedIdents.size());
+
+        Preconditions.checkArgument(
+            relatedEntityTypes.size() == 1,
+            "All entities must have the same related entity type, but got %s",
+            relatedEntityTypes.size());
+
+        Entity.EntityType type = relatedEntityTypes.iterator().next();
+        metaService.batchInsertStatisticPOsOnDuplicateKeyUpdate(
+            statisticEntities, relatedIdents.iterator().next(), type);
+        return null;
```

Comment:
Why do you need this?

Reply:
The treeLock can't accept the method return `void`. So I used a method return `Void`. I returned null here.